unified-runtime/source/adapters/opencl/command_buffer.cpp

```diff
@@ -115,6 +115,10 @@ urCommandBufferRetainExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
 UR_APIEXPORT ur_result_t UR_APICALL
 urCommandBufferReleaseExp(ur_exp_command_buffer_handle_t hCommandBuffer) {
   if (hCommandBuffer->decrementReferenceCount() == 0) {
+    if (hCommandBuffer->LastSubmission) {
```

Comment:
Might be that this is not needed either since opencl spec says: "After the command_buffer reference count becomes zero **and has finished execution**, the command-buffer is deleted."

https://registry.khronos.org/OpenCL/sdk/3.0/docs/man/html/clReleaseCommandBufferKHR.html